test/clone.test.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const rewire = require('rewire');
+const _cloneOptions = rewire('../').__get__('_cloneOptions');
+const http = require('http');
+const t = require('chai').assert;
+
+describe('Clone option function', function () {

```

Comment:
```
_cloneOptions(this.options)
```

output is available as the `retryStrategy` input. Instead of relying on a hack like `rewire` why not create a test, that specify :
-   an agent as option
-  its own retryStrategy, that we will use to check that  agent was not duplicated 

it should be a lot simpler, won't require rewire, have a simpler setup phase and only 2 assertions will be required (one for the agent the other for the other field that was cloned)